apps/github-api-integration/README.md

```diff
@@ -0,0 +1,212 @@
+API (Github) Integration
+========================
+
+This example demonstrates how to integrate a web form with a remote API using Home Office Forms (HOF).
+Specifically, this demonstrates how to integrate a HOF form with the Github API. However, this example only uses Github as the remote API in order to provide a potentially real-life example of API integration. With some small tweaks, pertaining to the url, and quite possibly the Model, this example could be repurposed to fit any number of simple API integrations.
+
+If you'd rather get started quickly [skip to install & start](#install-and-start)
+
+# Tutorial
+
+The three main components to this example are the Github model, the behaviours and the apps' `index.js` that configures the specfics of the app.
+
+To begin by writing the Model means when we introduce new components to the stack our upstream dependencies are known quantaties and therefore easier to reason about.
+
+## Models
+
+### Github model
+
+Our Github Model will extend from [hof-model](https://github.com/UKHomeOfficeForms/hof-model).
+```js
+const Model = require('hof-model');
+
+module.exports = class GithubModel extends Model {}
+```
+
+### Setting options at instantiation
+
+We know that [all Github API requests require a User-Agent header](https://developer.github.com/v3/#user-agent-required) so we can start by overriding the parent Model's `constructor` method to ensure all requests carry the correct header.
+
+```js
+constructor(attributes, options) {
+  // Call super first to create this.options
+  super(attributes, options);
+  // Set a custom header for the remote api
+  this.options.headers = {
+    'User-Agent': 'HOF Example API Integration'
+  };
+}
+```
+We can call `super` before setting the header to hand off the creation of `this.options` to the parent, base Model.
+
+### Setting the url
+
+The Model's `url` method will be called when any Model API method is called, but before the request to the Github API is invoked.
+In our example we override the parent Model's `url` method to return the `url` argument we passed to both the behaviours, replacing `:username` with the `name` entered by the user.
+
+```js
+url() {
+  // Augment configured url the user entered `name` param
+  // that we set on the model in the behaviour
+  return this.options.url.replace(':username', this.get('name'));
+}
+```
+
+When fetch is called from either behaviour, our Model's `url` method returns the url with which to make the request to the Github API.
+
+### Fetch Repos model
+
+Because our `show` view only requires a subset of the full response returned by the API, we extend from the github model, `require` the child model in the `fetch-repos` behaviour and parse the response.
+
+```js
+module.exports = class FetchReposModel extends Model {
+
+  // Parse the response from the
+  // API so we return only a subset of the results
+  parse(results) {
```

Comment:
I now understand that you're extending the parse function from the hof-model because you showed me.  Do you need to explain that in the docs to novices like myself?